src/environment.go

```diff
@@ -233,8 +233,8 @@ func (env *environment) getShellName() string {
 	if err != nil {
 		return unknown
 	}
-	shell := strings.Replace(name, ".exe", "", 1)
-	return strings.Trim(shell, " ")
+	*env.args.Shell = strings.Replace(name, ".exe", "", 1)
```

Comment:
I would set the name AFTER trimming it, no?
Meaning we do this:

```golang
shell := strings.Replace(name, ".exe", "", 1)
*env.args.Shell = strings.Trim(shell, " ")
return *env.args.Shell
```

Reply:
I thought the same but was not sure about what was the replace doing.